generics/generics_assignment.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"fmt"
+)
+
+type Person[T int | string] struct {
+	Name string
+	Age  T
+}
+
+func FindMax[V int | float64](slice []V) V {
+	slicecopied := []V{0}
+	for k, v := range slice {
```

Comment:
The `FindMax` function works well in principle, but the way the slice is copied could be simplified. Instead of creating a new slice slicecopied, you could use a variable to track the maximum value directly. This would simplify the code and avoid unnecessary memory allocation.
```suggestion
    for _, v := range slice[1:] {  // Start from the second element
```